### DISCUSSION TITLE
Data augmentation 재현성 관련 질문이 있습니다.

### DISCUSSION THREAD

**Discussion by ymnseol** [boostcampaitech4nlp2/level1_semantictextsimilarity_nlp-level1-nlp-13#20] [Q&A]
요약: data augmentation에서 잘 생성된 데이터만을 선별해서 train dataset에 추가하면 재현성 규칙 위반일까요?

현재 우리 모델의 문제는,
1. 모델이 단어 일치 정도를 통해 문장의 유사도를 측정해서, 문장의 의미를 제대로 학습하지 못하고,
2. 지나치게 긍정적이라는 것입니다.

문제1을 해결하기 위해, SR(Synonym Replacement)과 RI(Random Insertion)를 위주로 data augmentation을 수행하고자 합니다. KoEDA를 사용해 데이터를 증강할 수 있도록 합니다.
SR을 수행하던 중, 아래와 같은 문제를 맞닥뜨렸습니다. (아래 예시 문장들은 데이터셋 유출을 방지하기 위해, 비슷한 문제를 겪는 문장으로 대체했습니다.)

단어가 품사가 같지 않은 동의어로 대체되는 경우가 발생합니다.  
```
Original sentence: 오늘은 다행히 날씨가 맑네요!
Replaced sentence: 오늘백금 다행히 날씨가 맑네요!
```
문장에서 ‘은’은 조사이지만, 은(명사)의 동의어 백금(명사)으로 대체되었습니다.

또한, 동음이의어의 동의어로 대체되는 경우도 발생합니다.
```
Original sentence: 데이터 증강 건으로 질문 드립니다.
Replaced sentence: 데이터 증강 중화기으로 질문 드립니다.
```
원 문장에 쓰인 ‘건’을 기관총, 총, 중화기, 대포 등과 동의어라고 판단했습니다.

바로 위의 문제와 더불어, 잘못된 동의어 대체로 인해, 의도치 않게 문장의 감성이 완전히 뒤바뀌기도 합니다.
```
Original sentence: 그 분은 항상 1층 사무실에 계시더라
Replaced sentence: 그 똥은 항상 1층 사무실에 계시더라
```
이러한 문제를 해결하기 위해,
1. SR을 사용하지 않거나,
2. 문장 단어별 품사를 참고하여, 명사, 동사 등만 변경할 수 있도록 하거나,
3. 변경 대상이 되는 단어의 길이를 제한해, 상대적으로 길이가 짧은 품사 등은 최대한 변경하지 않도록 하거나,
4. 수동으로 좋은 변환 문장만을 선택해 사용합니다.

그래서 바뀐 단어 쭉 뽑아서 비교해서 잘 바뀐 애들만 학습 데이터에 추가하려고 했는데, 대회 룰 중에 
> [데이터 증강] Train/Dev 데이터에 한해 증강 가능합니다. 다만 기술적/통계적 근거가 존재해야하며, **누구나 동일한 증강 결과물을 생성할 수 있어야 합니다**.

가 있어서, 수동 선별을 하면 안 될지 궁금합니다!

**Comment by papari1123** [comment]
직접 핸드 라벨링해서 데이터를 증강하는 게 아니라, 기존 데이터에서 직접 추리는 거니깐 크게 문제는 안될 거 같은데..
주최측에 확인을 하는 게 좋을 듯.
다만, 추리는 과정에서도 로직이 있으면 좋겠지. 좋은 방법 중 하나는, 지금 개발된 모델로 일단 예측 결과를 얻고
예측값과 실제값 (diff)의 차이가 작은 것에 대해서 증강하는 거야.

물론 이 방법에 단점이 있는데, 기존에도 예측이 어려운 hard sample들은 증강이 안될 것이기 때문에, 데이터 증강 이후로도 결과가 나쁠 수 있겠지?

어쨌든 말한 내용들로 증강했을 때 어떻게 될지 기대가 되는군여

**Comment by intrandom5** [comment]
성구형이 좋은 방법 말해줬으니까... 그리고 단어들을 보니까 뭔가 라이브러리를 써서 하고 있는건가? 룰이 걱정되면 우리한테 주어진 데이터셋에 사용된 단어 중에서 골라 바꾸는 쪽으로 해도 괜찮을 듯.

**Comment by ymnseol** [comment]
정적 wordnet을 사용하지 않고 BERT를 이용하는 방향으로 SR을 수행해보겠습니다. 모두 조언 감사합니당!
